tests/test-team/template-mgmt-component-tests/template-mgmt-manage-templates-page.component.ts

```diff
@@ -0,0 +1,298 @@
+
+const templates = {
+  emailSubmitted: TemplateFactory.create({
+    id: uuid(),
+    version: 1,
+    name: 'email-submitted_manage-templates-page',
+    message: 'test example message',
+    subject: 'test example subject',
+    templateType: TemplateType.EMAIL,
+    templateStatus: TemplateStatus.SUBMITTED,
+    createdAt: '2010-10-11T11:11:11.111Z',
+  }),
+  emailNotYetSubmitted: TemplateFactory.create({
+    id: uuid(),
+    version: 1,
+    name: 'email-not-yet-submitted_manage-templates-page',
+    message: 'test example message',
+    subject: 'test example subject',
+    templateType: TemplateType.EMAIL,
+    templateStatus: TemplateStatus.NOT_YET_SUBMITTED,
+    createdAt: '2010-10-11T10:10:10.100Z',
+  }),
+  smsSubmitted: TemplateFactory.create({
+    id: uuid(),
+    name: 'sms-submitted_manage-templates-page',
+    message: 'test example message',
+    templateType: TemplateType.SMS,
+    templateStatus: TemplateStatus.SUBMITTED,
+    createdAt: '2010-10-10T11:11:11.111Z',
+  }),
+  smsNotYetSubmitted: TemplateFactory.create({
+    id: uuid(),
+    name: 'sms-not-yet-submitted_manage-templates-page',
+    message: 'test example message',
+    templateType: TemplateType.SMS,
+    templateStatus: TemplateStatus.NOT_YET_SUBMITTED,
+    createdAt: '2010-10-10T10:10:10.100Z',
+  }),
+  nhsAppSubmitted: TemplateFactory.create({
+    id: uuid(),
+    name: 'nhs-app-submitted_manage-templates-page',
+    message: 'test example message',
+    templateType: TemplateType.NHS_APP,
+    templateStatus: TemplateStatus.SUBMITTED,
+    createdAt: '2010-10-09T11:11:11.111Z',
+  }),
+  nhsAppNotYetSubmitted: TemplateFactory.create({
+    id: uuid(),
+    name: 'nhs-app-not-yet-submitted_manage-templates-page',
+    message: 'test example message',
+    templateType: TemplateType.NHS_APP,
+    templateStatus: TemplateStatus.NOT_YET_SUBMITTED,
+    createdAt: '2010-10-09T10:10:10.100Z',
+  }),
+};
+
+test.describe('Manage templates page', () => {
+  const templateStorageHelper = new TemplateStorageHelper(
+    Object.values(templates)
+  );
+
+  test.beforeAll(async () => {
+    await templateStorageHelper.seedTemplateData();
+  });
+
+  test.afterAll(async () => {
+    await templateStorageHelper.deleteTemplateData();
+  });
+
+  test('common page tests', async ({ page, baseURL }) => {
+    const props = {
+      page: new ManageTemplatesPage(page),
+      baseURL,
+    };
+
+    await assertSkipToMainContent(props);
+    await assertNotifyBannerLink(props);
+    await assertLoginLink(props);
+    await assertFooterLinks(props);
+    await assertGoBackLinkNotPresent(props);
+  });
+
+  test('should navigate to the manage templates page', async ({
+    page,
+    baseURL,
+  }) => {
+    const manageTemplatesPage = new ManageTemplatesPage(page);
+    await manageTemplatesPage.loadPage();
+    await expect(page).toHaveURL(`${baseURL}/templates/manage-templates`);
+    expect(await manageTemplatesPage.pageHeader.textContent()).toBe(
```

Comment:
Really minor.

This could be updated to

```
await expect(manageTemplatesPage.pageHeader).toHaveText('Message templates')
```

So the `await` is not nested into the expect. Generally if something bad happens with nested `awaits` the error is swallowed and harder to debug.

Playwright generally has an async matcher for most things.

Reply:
Change made.